pincer/objects/guild/guild.py

```diff
@@ -885,6 +890,366 @@ async def delete(self):
+            Reason for the prune |default| :data:`None`
+
+        Returns
+        -------
+        Dict[:class:`str`, :class:`int`]
+            An object with one 'pruned' key indicating
+            the number of members that were removed.
+        """
+        return await self._http.post(
+            f"guilds/{self.id}/prune",
+            data={
+                "days": days,
+                "compute_prune_days": compute_prune_days,
+                "include_roles": include_roles
+            },
+            headers={"X-Audit-Log-Reason": reason}
+            if reason is not None
+            else {}
+        )
+
+    async def get_voice_regions(self) -> AsyncGenerator[VoiceRegion, None]:
+        """|coro|
+        Returns an async generator of voice regions.
+
+        Returns
+        -------
+        AsyncGenerator[:class:`~pincer.objects.voice.VoiceRegion`, :data:`None`]
+            An async generator of voice regions.
+        """
+        data = await self._http.get(f"guilds/{self.id}/regions")
+        for voice_region_data in data:
+            yield VoiceRegion.from_dict(construct_client_dict(self._client, voice_region_data))
+
+    async def get_invites(self) -> AsyncGenerator[Invite, None]:
+        """|coro|
+        Returns an async generator of invites for the guild.
+        Requires the ``MANAGE_GUILD`` permission.
+
+        Returns
+        -------
+        AsyncGenerator[:class:`~pincer.objects.invite.Invite`, :data:`None`]
+            An async generator of invites.
+        """
+        data = await self._http.get(f"guilds/{self.id}/invites")
+        for invite_data in data:
+            yield Invite.from_dict(construct_client_dict(self._client, invite_data))
+
+    async def get_integrations(self) -> AsyncGenerator[Integration, None]:
+        """|coro|
+        Returns an async generator of integrations for the guild.
+        Requires the ``MANAGE_GUILD`` permission.
+
+        Returns
+        -------
+        AsyncGenerator[:class:`~pincer.objects.integration.Integration`, :data:`None`]
+            An async generator of integrations.
+        """
+        data = await self._http.get(f"guilds/{self.id}/integrations")
+        for integration_data in data:
+            yield Integration.from_dict(construct_client_dict(self._client, integration_data))
+
+    async def delete_integration(
+        self,
+        integration: Integration,
+        reason: Optional[str] = None
+    ):
+        """|coro|
+        Deletes an integration.
+        Requires the ``MANAGE_GUILD`` permission.
+
+        Parameters
+        ----------
+        integration : :class:`~pincer.objects.integration.Integration`
+            The integration to delete.
+        reason : Optional[:class:`str`]
+            Reason for the deletion |default| :data:`None`
+        """
+        await self._http.delete(
+            f"guilds/{self.id}/integrations/{integration.id}",
+            headers={"X-Audit-Log-Reason": reason}
```

Comment:
This line is getting repeat quite a few time, maybe we should consider having a headers object to simplify it?

> ex:
```py
headers.set_not_none("X-Audit-Log-Reason": reason)
```

Reply:
maybe maybe :thinking: